src/main/scala/diplomaticobjectmodel/logicaltree/RocketLogicalTree.scala

```diff
@@ -0,0 +1,58 @@
+package diplomaticobjectmodel.logicaltree
+
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.diplomacy.{LazyModule, ResourceBindingsMap, SimpleDevice}
+import freechips.rocketchip.diplomaticobjectmodel.model._
+import freechips.rocketchip.rocket.{DCacheParams, Frontend, ScratchpadSlavePort}
+import freechips.rocketchip.tile.{RocketTileParams, TileParams, XLen}
+
+
+class RocketLogicalTree(
+  device: SimpleDevice,
+  tileParams: TileParams,
+  rocketParams: RocketTileParams,
+  frontEnd: Frontend,
+  dtim_adapter: Option[ScratchpadSlavePort],
+  XLen: Int) extends LogicalTree {
+
+  def getOMDCacheFromBindings(dCacheParams: DCacheParams, resourceBindingsMap: ResourceBindingsMap): Option[OMDCache] = {
+    val omDTIM: Option[OMDCache] = dtim_adapter.map(_.device.getMemory(dCacheParams, resourceBindingsMap))
+    val omDCache: Option[OMDCache] = tileParams.dcache.filterNot(_.scratch.isDefined).map(OMCaches.dcache(_, None))
+
+    require(!(omDTIM.isDefined && omDCache.isDefined))
+
+    omDTIM.orElse(omDCache)
+  }
+
+  def getInterruptTargets(): Seq[OMInterruptTarget] = {
+    Seq(OMInterruptTarget(
+      hartId = rocketParams.hartId,
+      modes = OMModes.getModes(rocketParams.core.useVM)
+    ))
+  }
+
+  override def  getOMComponents(resourceBindingsMap: ResourceBindingsMap, components: Seq[OMComponent]): Seq[OMComponent] = {
```

Comment:
You pass in components as an argument and never use it.

Reply:
What I mean is, shouldn't `OMICache`, `OMDCache`, etc, be passed in here via `components: Seq[OMComponent]`? Otherwise what was the point of recursing through the tree?